src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -128,57 +128,57 @@ static bool TryParseValue(string value, out ImmutableHashSet<TEnum> result)
             return analyzerConfigOptions.GetOptionValue(optionName, rule, uint.TryParse, defaultValue);
         }
 
-        public static SymbolNamesOption GetNullCheckValidationMethodsOption(
+        public static SymbolNamesOption<byte> GetNullCheckValidationMethodsOption(
             this AnalyzerOptions options,
             DiagnosticDescriptor rule,
             Compilation compilation,
             CancellationToken cancellationToken)
-            => options.GetSymbolNamesOption(EditorConfigOptionNames.NullCheckValidationMethods, namePrefixOpt: "M:", rule, compilation, cancellationToken);
+            => options.GetSymbolNamesOption<byte>(EditorConfigOptionNames.NullCheckValidationMethods, namePrefixOpt: "M:", rule, compilation, cancellationToken);
 
-        public static SymbolNamesOption GetAdditionalStringFormattingMethodsOption(
+        public static SymbolNamesOption<byte> GetAdditionalStringFormattingMethodsOption(
             this AnalyzerOptions options,
             DiagnosticDescriptor rule,
             Compilation compilation,
             CancellationToken cancellationToken)
-            => options.GetSymbolNamesOption(EditorConfigOptionNames.AdditionalStringFormattingMethods, namePrefixOpt: "M:", rule, compilation, cancellationToken);
+            => options.GetSymbolNamesOption<byte>(EditorConfigOptionNames.AdditionalStringFormattingMethods, namePrefixOpt: "M:", rule, compilation, cancellationToken);
 
-        public static SymbolNamesOption GetExcludedSymbolNamesOption(
+        public static SymbolNamesOption<byte> GetExcludedSymbolNamesOption(
             this AnalyzerOptions options,
             DiagnosticDescriptor rule,
             Compilation compilation,
             CancellationToken cancellationToken)
-            => options.GetSymbolNamesOption(EditorConfigOptionNames.ExcludedSymbolNames, namePrefixOpt: null, rule, compilation, cancellationToken);
+            => options.GetSymbolNamesOption<byte>(EditorConfigOptionNames.ExcludedSymbolNames, namePrefixOpt: null, rule, compilation, cancellationToken);
 
-        public static SymbolNamesOption GetExcludedTypeNamesWithDerivedTypesOption(
+        public static SymbolNamesOption<byte> GetExcludedTypeNamesWithDerivedTypesOption(
             this AnalyzerOptions options,
             DiagnosticDescriptor rule,
             Compilation compilation,
             CancellationToken cancellationToken)
-            => options.GetSymbolNamesOption(EditorConfigOptionNames.ExcludedTypeNamesWithDerivedTypes, namePrefixOpt: "T:", rule, compilation, cancellationToken);
+            => options.GetSymbolNamesOption<byte>(EditorConfigOptionNames.ExcludedTypeNamesWithDerivedTypes, namePrefixOpt: "T:", rule, compilation, cancellationToken);
 
-        public static SymbolNamesOption GetDisallowedSymbolNamesOption(
+        public static SymbolNamesOption<byte> GetDisallowedSymbolNamesOption(
```

Comment:
I am wondering if the base type can be a non-generic type `SymbolNamesOption` and the sub-type can be `SymbolNamesOption<T>` so only the options which needs a key-value mapping will have to specify the type argument. What do you think?

Reply:
Probably also call it `SymbolNamesWithValuesOption<T>` or some such and move the `TryGetValue` function to this type.

Reply:
The rename is a good point. Regarding the base class not generic, I was thinking about that but I couldn't managed to do some implementation without too much duplication. Actually I did, but that was making the base handling a dictionary with `object` as value and so potentially forcing some unneeded boxing. Feel free to make a suggestion if you think I am missing something.

Reply:
Can we have an `AbstractSymbolNamesOption` which has all the core logic, with sub-types `SymbolNamesOption` and `SymbolNamesWithValueOption<T>` maintaining their required datastructures (set in the former, and dictionary for the latter)?

Reply:
We can but there won't be much code common between the 2, I'll add a separate commit so you can decide what you think is best.

Reply:
Ah, in that case it is fine to keep it as is.